crates/bevy_ecs/src/identifier/mod.rs

```diff
@@ -0,0 +1,103 @@
+//! A module for the unified [`Identifier`] ID struct, for use as a representation
+//! of multiple types of IDs in a single, packed type. Allows for describing an [`crate::entity::Entity`],
+//! or other IDs that can be packed and expressed within a `u64` sized type.
+//! [`Identifier`]s cannot be created directly, only able to be converted from other
+//! compatible IDs.
+use self::{kinds::IdKind, masks::IdentifierMask};
+
+pub mod error;
+pub(crate) mod kinds;
+pub(crate) mod masks;
+
+/// A unified identifier for all entity/component/relationship pair IDs.
```

Comment:
I'm a bit reluctant to ship doc strings that mention relationships before they're added.